weekly_tails.py

```
@@ -568,43 +575,169 @@ def _check_tail_above_fibonacci_resistance(self, tail_price: float, fib_levels:
+                        weighted_tail['trend_adjustment'] = 'BEAR_REDUCED'
+                        weighted_tail['reason'] = f"{tail.get('reason', '')} (Reduced confidence in bear market)"
+                
+                else:  # NEUTRAL
+                    weighted_tail['trend_adjustment'] = 'NO_ADJUSTMENT'
+                
+                # Update strength category after weighting
+                weighted_tail['strength_category'] = self._categorize_strength(weighted_tail['strength'])
+                weighted_tails.append(weighted_tail)
+            
+            logger.info(f"Applied trend weighting: {market_trend} market")
+            return weighted_tails
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)